### ISSUE TITLE
bug(mat-menu): In high contrast mode, borders not fully visible around mat-menu-items

### ISSUE THREAD

**Issue by dwaynelavon** [angular/components#21524]
#### Reproduction

Stackblitz: https://stackblitz.com/angular/ooebrkdoyjb?file=src/app/menu-overview-example.html

Steps to reproduce:
1. Enable Windows high contrast mode in settings
2. Open firefox
3. Create a mat-menu with 3+ options (Stackblitz above)
4. Open the menu and navigate through the options
 
 
#### Expected Behavior

What behavior were you expecting to see?
Both the top and bottom borders should be visible around menu items. 

#### Actual Behavior

What behavior did you actually see?
Only the last item shows both borders. The other items only show one border when active.

#### Environment

- Angular: 11.0.3
- CDK/Material: 11.0.3
- Browser(s): Firefox
- Operating System (e.g. Windows, macOS, Ubuntu):  Windows

![image](https://user-images.githubusercontent.com/11365475/104066053-5c5e7200-51c6-11eb-8513-2a93220d418f.png)
